components/fri3d_bsp/src/boards/fox/bsp_button.c

```diff
@@ -0,0 +1,47 @@
+#include "esp_log.h"
+#include "fri3d_bsp/bsp.h"
+#include "iot_button.h"
+
+// on create replaced by dynamic version
+static bsp_button_custom_config_t bsp_custom_button_boot_config = {
+    .gpio_num = BSP_BUTTON_BOOT_IO,
+    .active_level = 0,
+    .gpio_pull_mode = GPIO_FLOATING
+};
+
+
+const button_config_t bsp_button_config[BSP_BUTTON_NUM] = {
+    {
+        .type = BUTTON_TYPE_CUSTOM,
+        .custom_button_config.active_level = 0,
+        .custom_button_config.button_custom_init = bsp_button_custom_init,
+        .custom_button_config.button_custom_get_key_value = bsp_button_custom_get_key_level,
+        .custom_button_config.button_custom_deinit = NULL,
+        .custom_button_config.priv = &bsp_custom_button_boot_config
```

Comment:
original struct says: `/**< private data used for custom button, MUST be allocated dynamically and will be auto freed in iot_button_delete*/`
is not dynamically allocated here, but this is later on replaced see `components/fri3d_bsp/src/bsp_button.c`

Reply:
solved with custom struct to define the button configuration